### PR TITLE
Fix flakiness in dynamic mapping update tests

### PR DESCRIPTION
**Pull request by BaurzhanSakhariev**
Wait for mapping update on all nodes before selecting. Similar to https://github.com/crate/crate/commit/d41c5a1721d1c2f6c77f13f3ad6b5d587978123f

### CODE REVIEW

**Review by matriv — APPROVED** [review]

File: server/src/test/java/io/crate/integrationtests/SQLTypeMappingTest.java [hunk 1]
```diff
@@ -184,6 +183,8 @@ public void testInvalidInsertIntoObject() throws Exception {
             .hasPGError(INTERNAL_ERROR)
             .hasHTTPError(BAD_REQUEST, 4000)
             .hasMessageContaining("Cannot cast object element `created` with value `true` to type `timestamp with time zone`");
+
+        waitNoPendingTasksOnAll();
```
Comment (seut) [inline server/src/test/java/io/crate/integrationtests/SQLTypeMappingTest.java:187]: What will this help here? Either the assertion will kick in or the test will fail anyhow

File: server/src/test/java/io/crate/integrationtests/SQLTypeMappingTest.java [hunk 2]
```diff
@@ -347,6 +349,8 @@ public void testInsertNewColumnToStrictObject() throws Exception {
             .hasPGError(INTERNAL_ERROR)
             .hasHTTPError(BAD_REQUEST, 4000)
             .hasMessageContaining("Cannot add column `another_new_col` to strict object `strict_field`");
+
+        waitNoPendingTasksOnAll();
```
Comment (seut) [inline server/src/test/java/io/crate/integrationtests/SQLTypeMappingTest.java:353]: Same here, what will be solved by this?

**Comment by seut** [comment]
Are we sure we are looking at the right issue?
If an write will result in new columns, the `ShardUpsertAction` logic will do a blocking call to add the missing columns.
This request is an "acked" request, meaning it only returns once all nodes consumed the new cluster state and send a response. 

The mentioned commit https://github.com/crate/crate/commit/d41c5a1721d1c2f6c77f13f3ad6b5d587978123f is quite old, done when we didn't have a custom `add-column-blocking` logic.

**Comment by BaurzhanSakhariev** [comment]
> This request is an "acked" request, meaning it only returns once all nodes consumed the new cluster state and send a response.

Makes sense, closing this

**Pull request closed without merging**